Providers/PythonProvider.cpp

```diff
@@ -48,22 +49,51 @@ char const SCRIPT_PATH_EXTENSION[] = "/lib/Scripts/";
 char const DEFAULT_DSC_SCRIPT[] = "client";
 char const PY_EXTENSION[] = ".py";
 
+std::string determinePythonVersion(){
+    std::array<char, 128> buffer;
+    std::string result;
+    // Check for python2
+    FILE* pipe = popen("python --version 2>&1", "r");
```

Comment:
still would suggest using the command "python2" instead of unversioned "python"
imagine scenario where customer is on RHEL8 and only installs python2. it isn't aliased to python by default. no python will be found